lib/matplotlib/image.py

```diff
@@ -486,16 +486,30 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
                     raise ValueError("Invalid dimensions, got %s" % (A.shape,))
 
                 output = np.zeros((out_height, out_width, 4), dtype=A.dtype)
+                output_a = np.zeros((out_height, out_width, 4), dtype=A.dtype)
 
                 alpha = self.get_alpha()
                 if alpha is None:
-                    alpha = 1.0
+                    alpha = 1
+
+                #resample alpha channel
+                alpha_tuple = (A[...,3], A[...,3], A[...,3])
+                alpha_channel = _rgb_to_rgba(np.dstack(alpha_tuple))
+                _image.resample(
+                    alpha_channel, output_a, t, _interpd_[self.get_interpolation()],
+                    self.get_resample(), alpha,
+                    self.get_filternorm(), self.get_filterrad())
 
+                #resample rgb channels
+                A = _rgb_to_rgba(A[...,:3])
                 _image.resample(
                     A, output, t, _interpd_[self.get_interpolation()],
                     self.get_resample(), alpha,
                     self.get_filternorm(), self.get_filterrad())
 
+                #recombine rgb and alpha channels
+                output = np.dstack((output[...,:3], output_a[...,0]))
```

Comment:
Can we update `output` in place here?  Memory allocation in this code path is a mid-level concern.